File: src/main/java/com/java/scogen/dataGenerator/EmployeeDetailsDataset.java
package com.java.scogen.dataGenerator;

import com.java.scogen.pojo.EmployeeDetailsPojo;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public class EmployeeDetailsDataset {

    private SparkSession spark;

    public EmployeeDetailsDataset(SparkSession spark) {
        this.spark= spark;
    }

    public Dataset<Row> EmployeeTable(Set<String> nameSet)
    {
        List<EmployeeDetailsPojo> namesDetails = nameSet.stream().map(EmployeeDetailsPojo::new )
                .collect(Collectors.toList());

        return spark.createDataFrame(namesDetails,EmployeeDetailsPojo.class);
    }
}

File: src/main/scala/com/scala/scogen/datagenerator/EmployeeDetailsDataset.scala
package com.scala.scogen.datagenerator

import com.scala.scogen.caseclasses.EmployeeDetailsPojo
import org.apache.spark.sql.{Dataset, Row, SparkSession}


class EmployeeDetailsDataset {

  def employeeTable(spark:SparkSession,nameSet: Set[String]): Dataset[Row] = {
    import spark.implicits._
    val namesDetails = nameSet.map(x=>EmployeeDetailsPojo(x)).toList
    spark.sparkContext.parallelize(namesDetails).toDF()
  }

}
